src/nupic/torch/modules/sparse_weights.py

```diff
@@ -134,6 +138,18 @@ class SparseWeights(SparseWeightsBase):
     def __init__(self, module, weight_sparsity=None, sparsity=None,
                  allow_extremes=False):
         assert len(module.weight.shape) == 2, "Should resemble a nn.Linear"
+
+        # Add checks to ensure there is no incompatibility
+        if sparsity is None:
+            sparsity = 1 - weight_sparsity
+        if weight_sparsity is None:
+            weight_sparsity = 1 - sparsity
+        if sparsity is not None and weight_sparsity is not None:
+            assert (1 - sparsity) == weight_sparsity, (
+                "Conflicting sparsity levels defined by the params weight "
```

Comment:
This assert belongs to `SparseWeightsBase` covering both linear and conv2d sparse layers